examples/tcp-retransmit.yaml

```diff
@@ -0,0 +1,56 @@
+metrics:
+    counters:
+      - name: tcp_retransmit_ipv4_count
+        help: Total count of IPv4 TCP retransmissions
+        labels:
+          - name: src_ip
```

Comment:
Having IP and port is indeed too much cardinality for something one could run normally in production. It might be more useful to limit to just port and maybe an interface name.

It's probably not too useful to use the whole port range either. A possible heuristic is to:

1. Pick a min port number between src and dst. E.g. you'll get port 443 for both incoming and outgoing packets, no matter if this machine acts like a client or a server. It's probably best to call this label `main_port`, `min_port` or something like it, so that it doesn't clash with src or dst semantics.
2. Ignore events if the resulting port is in the ephemeral range: [examples/udp-drops.bpf.c#L5](https://github.com/cloudflare/ebpf_exporter/blob/fe73d1c09fe10/examples/udp-drops.bpf.c#L5).

Reply:
Do you find it useful?

I'm not sure that the port data will be very useful and/or not misleading given that the `net.ipv4.ip_local_port_range` parameter is configurable.
(this is in addition to adding cardinality)

Reply:
The repo contains _examples_, they are not meant to be set in stone. If somebody adjust `net.ipv4.ip_local_port_range`, they can definitely adjust a `define` in the bpf source code as well.

Reply:
I left only `main_port` and renamed the `src_ip`, `dst_ip` labels to `local_ip` and `peer_ip`.